### PR TITLE
Convert to range-based for loops

### PR DESCRIPTION
**Pull request by gnawme**
In #5397 @mvieth suggested that iterations through containers be converted to range-based for loops as part of the modernization process.

This PR splits off from #5397 to make that PR smaller and easier to review.

### CODE REVIEW

**Comment by mvieth** [comment]
TBH I don't really see any benefit in converting for-loops that need a counter/iterator variable like `i` in [apps/src/openni_feature_persistence.cpp](https://github.com/PointCloudLibrary/pcl/pull/5417/files#diff-b257e68fa44f8c7bf798b8a45455670b1b93dc3d59cc274338256e2d341d3562). I don't like that the variable is accessible outside of the loop. I found at least one instance ([apps/src/feature_matching.cpp](https://github.com/PointCloudLibrary/pcl/pull/5417/files#diff-12a9df58c4c97d684f3dda71328f2e985d03d5f5ad50dcd42d22809f7c0e28c9)) where this lead to a bug.
Let's maybe convert only those for-loops that use begin() and end(), and those that can work without a counter/iterator `i` as in [doc/tutorials/content/sources/iros2011/src/build_all_object_models.cpp](https://github.com/PointCloudLibrary/pcl/pull/5417/files#diff-90aab5bf0082888d965e1b5c6fa92cf2f438e7ea92ab4fcc3e11432cdbe58f1c) ?

**Comment by gnawme** [comment]
> TBH I don't really see any benefit in converting for-loops that need a counter/iterator variable like `i` in [apps/src/openni_feature_persistence.cpp](https://github.com/PointCloudLibrary/pcl/pull/5417/files#diff-b257e68fa44f8c7bf798b8a45455670b1b93dc3d59cc274338256e2d341d3562). I don't like that the variable is accessible outside of the loop. I found at least one instance ([apps/src/feature_matching.cpp](https://github.com/PointCloudLibrary/pcl/pull/5417/files#diff-12a9df58c4c97d684f3dda71328f2e985d03d5f5ad50dcd42d22809f7c0e28c9)) where this lead to a bug. Let's maybe convert only those for-loops that use begin() and end(), and those that can work without a counter/iterator `i` as in [doc/tutorials/content/sources/iros2011/src/build_all_object_models.cpp](https://github.com/PointCloudLibrary/pcl/pull/5417/files#diff-90aab5bf0082888d965e1b5c6fa92cf2f438e7ea92ab4fcc3e11432cdbe58f1c) ?

I have to agreed; reverted selected modifications.

File: gpu/features/test/test_normals.cpp [hunk 1]
```diff
@@ -77,11 +77,11 @@ TEST(PCL_FeaturesGPU, normals_lowlevel)
     std::vector<PointXYZ> downloaded;
     normals.download(downloaded);
 
-    for(std::size_t i = 0; i < downloaded.size(); ++i)
+    for(const auto& download : downloaded)
```
Comment (mvieth) [inline gpu/features/test/test_normals.cpp:80]: `i` is needed in each of these for-loops to access the normals

File: gpu/features/test/test_fpfh.cpp [hunk 2]
```diff
@@ -90,9 +90,9 @@ TEST(PCL_FeaturesGPU, fpfh_low_level)
     PointCloud<FPFHSignature33> fpfhs;
     fe.compute (fpfhs);
 
-    for(std::size_t i = 0; i < downloaded.size(); ++i)
+    for(const auto& download : downloaded)
```
Comment (mvieth) [inline gpu/features/test/test_fpfh.cpp:93]: `i` is needed here to access `fpfhs` (every for-loop)

File: recognition/src/ransac_based/obj_rec_ransac.cpp [hunk 3]
```diff
@@ -42,180 +42,195 @@
 
 using namespace pcl::common;
 
-pcl::recognition::ObjRecRANSAC::ObjRecRANSAC (float pair_width, float voxel_size)
```
Comment (mvieth) [inline recognition/src/ransac_based/obj_rec_ransac.cpp:45]: Please undo all the (formatting) changes in this file, let's keep the PR on topic

File: apps/modeler/src/scene_tree.cpp [hunk 4]
```diff
@@ -384,23 +378,18 @@ pcl::modeler::SceneTree::slotUpdateOnSelectionChange(const QItemSelection& selec
...
   QModelIndexList deselected_indices = deselected.indexes();
-  for (QModelIndexList::const_iterator deselected_indices_it =
-           deselected_indices.begin();
-       deselected_indices_it != deselected_indices.end();
-       ++deselected_indices_it) {
-    QTreeWidgetItem* item = itemFromIndex(*deselected_indices_it);
-    RenderWindowItem* render_window_item = dynamic_cast<RenderWindowItem*>(item);
+  for (const auto& deselected_index : deselected_indices) {
+    QTreeWidgetItem* item = itemFromIndex(deselected_index);
+    auto render_window_item = dynamic_cast<RenderWindowItem*>(item);
```
Comment (mvieth) [inline apps/modeler/src/scene_tree.cpp:392]: Let's use `auto*` here

File: recognition/src/ransac_based/obj_rec_ransac.cpp (hunk 3 above)
Comment (gnawme) [reply recognition/src/ransac_based/obj_rec_ransac.cpp]: It's a mystery to me as to what changes `clang-format` will make to particular files in this code base.

**Review by larshg — APPROVED** [review]

**Review by mvieth — APPROVED** [review]
Thank you!

**Pull request merged**